src/server/amqp.js

```diff
@@ -63,11 +64,23 @@ export function getNotifications(user, ws) {
             config.amqpExchangeName,
             NOTIFICATION_ROUTING_KEY + user
         );
-        logger.info("Websocket closed. Channel unbound for user: " + user);
+        msgChannel.cancel(consumerTag);
+        logger.info(
+            "Websocket closed. Channel unbound for user: " +
+                user +
+                " consumer Tag:" +
+                consumerTag
+        );
     });
 
     ws.on("error", function (error) {
         logger.info("Websocket error: " + error);
+        msgChannel.unbindQueue(
+            QUEUE,
+            config.amqpExchangeName,
+            NOTIFICATION_ROUTING_KEY + user
+        );
+        msgChannel.cancel(consumerTag);
```

Comment:
Minor suggestion, but since this logic is repeated in `ws.on("close",...)`, I would have put them into some kind of "connectionCleanup" function, and let both handlers call that.